agent/engine/execcmd/agent_version.go

```diff
@@ -15,68 +15,107 @@ package execcmd
 
 import (
 	"errors"
+	"io/ioutil"
 	"strconv"
 	"strings"
 )
 
-// TODO: Remove this later, just doing it do make static checks happy about these utils not being invoked anywhere
-func init() {
-	determineLatestVersion([]string{"1.1.1.1"})
-}
-
 const (
 	versionDelimiter  = "."
-	badInputReturn    = -999
 	badFormatErrorStr = "invalid version format"
 )
 
-// compareAgentVersion compares two versions, returning -1 if v1 < v2, 0 if v1 == v2, and 1 if v1 > v2.
-// The version must have only integers representing major, minor, patch, and reserved; delimited by dots (e.g. "1.2.3.4").
-// An error is returned if any of the versions passed as parameter is malformed.
-// example:
-// compareAgentVersion("3.0.236.0", "3.0.237.0") returns -1, nil
-// compareAgentVersion("3.0.236.0", "3.0.236.0") returns 0, nil
-// compareAgentVersion("3.0.237.0", "3.0.236.0") returns 1, nil
-// compareAgentVersion("a.b.c.d", "3.0.236.0") returns -999, error
-func compareAgentVersion(v1Str, v2Str string) (int, error) {
-	v1Arr := strings.Split(v1Str, versionDelimiter)
-	v2Arr := strings.Split(v2Str, versionDelimiter)
+type agentVersion []uint
 
-	if len(v1Arr) != len(v2Arr) {
-		return badInputReturn, errors.New(badFormatErrorStr)
+// compare compares this agentVersion against another one passed as parameter.
+// returns -1 if this < av2, 0 this == av2, and 1 if this > av2.
+// example:
+// [3, 0, 236, 0].compare([3, 0, 237, 0]) returns -1
+// [3, 0, 236, 0].compare([3, 0, 236, 0]) returns 0
+// [3, 0, 237, 0].compare([3, 0, 236, 0]) returns 1
+// [3].compare([2, 0, 236])               returns 1
+func (av agentVersion) compare(av2 agentVersion) int {
```

Comment:
nit: `lhs` and `rhs` for compare functions

Reply:
I like that! will change